### PR TITLE
Delete always true conditions with version compare

### PR DESCRIPTION
**Pull request by nokitakaze**
Small change. Delete dead code

### CODE REVIEW

File: tests/ClientTest.php [hunk 1]
```diff
@@ -1846,15 +1836,7 @@ public function testEncodeTooDepth()
...
-        if (version_compare(PHP_VERSION, '5.5.0', '>=')) {
-            $this->assertFalse($value, 'Broken data encoded successfully with native method');
-        } else {
-            if ($value !== false) {
-                $this->markTestSkipped();
-            } else {
-                $this->assertEquals('eJyLjh4Fo2AUjFgQOwpGwSgYuQAA3Q7g1w==', $value, 'Native json_encode error');
-            }
-        }
+        $this->assertFalse($value, 'Broken data encoded successfully with native method');
```
Comment (ste93cry) [inline tests/ClientTest.php]: Didn't this removed too much code? I'm not sure the `else` part should be entirely removed

Comment (nokitakaze) [inline tests/ClientTest.php]: `version_compare(PHP_VERSION, '5.5.0', '>=')` is always true condition

File: tests/ClientTest.php [hunk 2]
```diff
@@ -1838,25 +1828,6 @@ public function test_server_variable()
         }
     }
 
-    public function testEncodeTooDepth()
```
Comment (nokitakaze) [inline tests/ClientTest.php]: I deleted the entire method because we do not need to test INTERNAL PHP FUNCTIONS!

Comment (nokitakaze) [inline tests/ClientTest.php]: @ste93cry, I need your opinion. Do you approve this?

Comment (ste93cry) [inline tests/ClientTest.php]: Yes it's fine

**Pull request merged**